### PR TITLE
ci: perftool for whole repo, fail on bad changes

### PR DESCRIPTION
**Pull request by akhdrv**
:baby_chick: Download canary assets:

[colors--canary.418.4662462519.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.418.4662462519.xml)  
[color_sberHealth_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.418.4662462519.swift)  
[color_sberHealth_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.418.4662462519.kt)  
[color_sberHealth_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.418.4662462519.ts)  
[color_sbermarket_business_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.418.4662462519.swift)  
[color_sbermarket_business_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.418.4662462519.kt)  
[color_sbermarket_business_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.418.4662462519.ts)  
[color_sbermarket_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.418.4662462519.swift)  
[color_sbermarket_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.418.4662462519.kt)  
[color_sbermarket_metro_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.418.4662462519.swift)  
[color_sbermarket_metro_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.418.4662462519.kt)  
[color_sbermarket_metro_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.418.4662462519.ts)  
[color_sbermarket_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.418.4662462519.ts)  
[color_sbermarket_selgros_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.418.4662462519.swift)  
[color_sbermarket_selgros_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.418.4662462519.kt)  
[color_sbermarket_selgros_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.418.4662462519.ts)  
[color_sbermarket_vinlab_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_vinlab_ios-swift--canary.418.4662462519.swift)  
[color_sbermarket_vinlab_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_vinlab_kotlin--canary.418.4662462519.kt)  
[color_sbermarket_vinlab_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_vinlab_react-native--canary.418.4662462519.ts)  
[color_sberprime_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.418.4662462519.swift)  
[color_sberprime_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.418.4662462519.kt)  
[color_sberprime_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.418.4662462519.ts)  
[PlasmaTokensColor--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.418.4662462519.swift)  
[shadow_sbermarket_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.418.4662462519.ts)  
[typo_mage_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.418.4662462519.swift)  
[typo_mage_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.418.4662462519.kt)  
[typo_mage_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.418.4662462519.ts)  
[typo_plasma_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.418.4662462519.swift)  
[typo_plasma_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.418.4662462519.kt)  
[typo_plasma_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.418.4662462519.ts)  
[typo_ruler_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.418.4662462519.swift)  
[typo_ruler_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.418.4662462519.kt)  
[typo_ruler_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.418.4662462519.ts)  
[typo_sage_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.418.4662462519.swift)  
[typo_sage_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.418.4662462519.kt)  
[typo_sage_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.418.4662462519.ts)  
[typo_sbermarket_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.418.4662462519.swift)  
[typo_sbermarket_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.418.4662462519.kt)  
[typo_sbermarket_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.418.4662462519.ts)  
[typo_soulmate_ios-swift--canary.418.4662462519.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.418.4662462519.swift)  
[typo_soulmate_kotlin--canary.418.4662462519.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.418.4662462519.kt)  
[typo_soulmate_react-native--canary.418.4662462519.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.418.4662462519.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-temple@1.149.2-canary.418.4662462519.0
  npm install @salutejs/plasma-ui@1.181.2-canary.418.4662462519.0
  # or 
  yarn add @salutejs/plasma-temple@1.149.2-canary.418.4662462519.0
  yarn add @salutejs/plasma-ui@1.181.2-canary.418.4662462519.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-418/

File: perftool.config.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ const config: Config = {
     include: ['src/**/*.perftest.tsx'],
     outputFilePath: 'perftest/result.json',
     displayIntermediateCalculations: false,
-    failOnSignificantChanges: false,
+    failOnSignificantChanges: true,
     stabilizers: ['staticTask'],
     absoluteError: 1,
     modifyWebpackConfig: function (config) {
```
Comment (Yeti-or) [inline perftool.config.ts:25]: что-то падает @akhdrv  это норм?

Comment (akhdrv) [reply perftool.config.ts:25]: пока не норм. еще правлю

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-418/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-418/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-418/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-418/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-418/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-418/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-418/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-418/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-418/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-418/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-418/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-418/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-418/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-418/temple-storybook/

File: perftool.config.ts [hunk 2]
```diff
@@ -16,25 +16,27 @@ const config: Config = {
...
+            babelLoaderOpts?.options?.plugins?.push(['const-enum', { transform: 'constObject' }]);
         }
 
         (config.resolve = config.resolve || {}).alias = {
-            react: '/node_modules/react',
-            'react-dom': '/node_modules/react-dom',
-            'styled-components': '/node_modules/styled-components',
+            react: '/packages/plasma-ui/node_modules/react',
+            'react-dom': '/packages/plasma-ui/node_modules/react-dom',
+            'styled-components': '/packages/plasma-ui/node_modules/styled-components',
```
Comment (neretin-trike) [inline perftool.config.ts:39]: а эти резолвы справедливы только для `plasma-ui`?

Comment (akhdrv) [reply perftool.config.ts:39]: это чтобы дедуплицировать пакеты после линкинга лерны, иначе при сборке в приложение попадает несколько реактов и styled-components

Comment (neretin-trike) [reply perftool.config.ts:39]: да, это я понял. Но тут больше вопрос к тому, что не нужно ли тоже самое для plasma-hope / core / web / b2c?

Comment (akhdrv) [reply perftool.config.ts:39]: не нужно

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**